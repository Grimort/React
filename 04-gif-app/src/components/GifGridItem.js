import React from "react";

const GifGridItem = ({ id, title, url }) => {
  return (
    <div className="card animate__animated animate__bounce animate__fadeInDown">
      <p>{title}</p>
      <img src={url.url} alt={title}></img>
    </div>
  );
};

export default GifGridItem;
