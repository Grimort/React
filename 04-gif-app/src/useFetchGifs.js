import { useState, useEffect } from "react";
import { GetGifs } from "./helpers/GetGifs";

export const useFetchGifs = (category) => {
  const [state, setState] = useState({
    data: [],
    loading: true,
  });

  useEffect(() => {
    setTimeout(() => {
      GetGifs(category).then((imgs) => {
        setState({
          data: imgs,
          loading: false,
        });
      });
    }, 1000);
  }, [category]);

  return state;
};
