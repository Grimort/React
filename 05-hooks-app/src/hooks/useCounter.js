import { useState } from "react";

export const useCounter = (initialState = 10) => {
  const [state, setState] = useState(initialState);
  const increase = (factor = 1) => {
    setState(state + factor);
  };
  const decrease = (factor = 1) => {
    setState(state - factor);
  };

  const reset = () => {
    setState(initialState);
  };
  return { state, increase, decrease, reset };
};
