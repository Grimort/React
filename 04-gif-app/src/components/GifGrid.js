import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";
import PropTypes from "prop-types";

export const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3 className="animate__animated animate__fadeIn" key={category}>
        {category}
      </h3>
      <div className="card-grid animate__animated animate__bounce animate__fadeIn">
        {loading && (
          <p className="animate__animated animate__flash">Loading...</p>
        )}
        {images.map((img) => (
          <GifGridItem key={img.id} url={img.url.url} title={img.title} />
        ))}
      </div>
    </>
  );
};
GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
