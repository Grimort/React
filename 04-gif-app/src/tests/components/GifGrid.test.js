import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";

jest.mock("../../hooks/useFetchGifs");

describe("Testing GifGrid component", () => {
  const category = "Memes";
  let wrapper;
  beforeEach(() => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    wrapper = shallow(<GifGrid category={category} />);
  });
  test("should to show the component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("shouldn't show items", () => {
    const gifs = [];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    wrapper = shallow(<GifGrid category="" />);
    expect(wrapper.find("p").exists()).toBeFalsy();
  });
  test("should show items when images has been loaded", () => {
    const gifs = [
      {
        id: "id-1",
        title: "title-1",
        url: "url-1",
      },
      {
        id: "id-2",
        title: "title-2",
        url: "url-2",
      },
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: true,
    });
    expect(wrapper.find("p").exists()).toBeTruthy();
  });
});
