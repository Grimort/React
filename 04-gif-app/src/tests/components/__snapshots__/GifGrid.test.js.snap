// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifGrid component should to show the component correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
    key="Memes"
  >
    Memes
  </h3>
  <div
    className="card-grid animate__animated animate__bounce animate__fadeIn"
  >
    <p
      className="animate__animated animate__flash"
    >
      Loading...
    </p>
  </div>
</Fragment>
`;
